tf2/src/buffer_core.cpp

```diff
@@ -182,27 +182,27 @@ bool BufferCore::setTransform(
   const geometry_msgs::msg::TransformStamped & transform,
   const std::string & authority, bool is_static)
 {
-  tf2::Transform tf2_transform(tf2::Quaternion(
+  tf2::Quaternion rotation(
```

Comment:
Note for the next reviewer, this change keeps the rotation quat as a quat and the positions as a Vec3 instead of promoting it to a tf2::Transform, saving us a slight conversion cost.

Reply:
I can actually just add these as comments